src/contracts/rewardsV2/ProtocolFees.sol

```diff
@@ -0,0 +1,191 @@
+pragma solidity 0.8.25;
+
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {IProtocolFees} from "../../interfaces/rewardsV2/IProtocolFees.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+/**
+ * @title ProtocolFees
+ * @notice Abstract contract that manages protocol fees for different reward types
+ * @dev This contract provides a base implementation for protocol fee management
+ * with support for different reward types and networks
+ */
+abstract contract ProtocolFees is OwnableUpgradeable, IProtocolFees {
+    using SafeERC20 for IERC20;
+    using Math for uint256;
+
+    /* CONSTANTS */
+
+    uint256 internal constant MAX_FEE = 1_000_000;
+
+    // keccak256(abi.encode(uint256(keccak256("protocol.fees.storage")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant PROTOCOL_FEES_STORAGE_POSITION =
+        0x6089e4218b2a24d7083b5793f90f138cd4eb34fc509b0fca031795f497243e00;
+
+    /**
+     * @notice Storage structure for protocol fees
+     */
+    struct ProtocolFeesStorage {
+        mapping(uint64 rewardsType => uint256 value) _fee;
+        // value is enable | fee (packed as bytes32)
+        mapping(uint64 rewardsType => mapping(address network => bytes32 value)) _networkFee;
+        mapping(address token => uint256 fee) _claimableFee;
+    }
+
+    /* FUNCTIONS */
+
+    /**
+     * @notice Initialize the protocol fees contract
+     * @param initParams Initialization parameters containing fee configurations
+     */
+    function __ProtocolFees_init(
+        ProtocolFeesInitParams calldata initParams
+    ) internal onlyInitializing {
+        __Ownable_init(msg.sender);
+
+        for (uint256 i = 0; i < initParams.fees.length; ++i) {
+            if (initParams.fees[i].fee > MAX_FEE) {
+                revert FeeTooHigh();
+            }
+            _protocolFeesStorage()._fee[initParams.fees[i].rewardsType] = initParams.fees[i].fee;
+        }
+    }
+
+    /**
+     * @notice Get the claimable protocol fees for a token
+     * @param token The token address
+     * @return The claimable fee amount
+     */
+    function claimableProtocolFees(
+        address token
+    ) public view returns (uint256) {
+        return _protocolFeesStorage()._claimableFee[token];
+    }
+
+    /**
+     * @notice Get the protocol fee for a reward type and network
+     * @param rewardsType The reward type identifier
+     * @param network The network address
+     * @return The protocol fee amount
+     */
+    function protocolFee(uint64 rewardsType, address network) public view returns (uint256) {
+        bytes32 networkFeeData = _protocolFeesStorage()._networkFee[rewardsType][network];
+
+        // Check if network fee is enabled
+        if (networkFeeData != bytes32(0)) {
```

Comment:
Seems (bool isEnabled, uint256 fee) = _deserializeNetworkFeeData(networkFeeData); can be done without this check